### PR TITLE
Good set of improvements for Blockly

### PR DESCRIPTION
**Pull request by alecont**
see the various commits for more details: more blocks, more samples, event handling, more Sense Hat features coming online (joystick, sensors, etc.)

### CODE REVIEW

File: IoTBlockly/IoTBlocklyHelper/Gpio.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public void SetGPIOPin(int pinNumber, int value)
             }
         }
 
-        private GpioPin GetPin(int pinNumber)
+        private static GpioPin GetPin(int pinNumber)
```
Comment (bfjelds) [inline IoTBlockly/IoTBlocklyHelper/Gpio.cs:60]: shouldn't the usage of GetPin switch from <instance>.GetPin to <class>.GetPin?  I don't see a corresponding change?

File: IoTBlockly/IoTBlocklyBackgroundApp/StartupTask.cs [hunk 2]
```diff
@@ -31,20 +31,107 @@ public sealed class StartupTask : IBackgroundTask
...
+            //";
+
+            //            var code = @"
+            //var basic = new IoTBlocklyHelper.Basic();
+            //var gpio = new IoTBlocklyHelper.Gpio();
+            //var senseHat = new IoTBlocklyHelper.SenseHat();
+            //var adc = new IoTBlocklyHelper.Adc();
+            //var eventsQueue = [];
+
+            //function pauseHelper(ms) {
```
Comment (bfjelds) [inline IoTBlockly/IoTBlocklyBackgroundApp/StartupTask.cs]: is pauseHelper defined elsewhere?  I see it used in active code below and commented out here?

File: IoTBlockly/IoTBlocklyBackgroundApp/public/blockly/javascript-custom.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ Blockly.JavaScript['device_forever'] = function(block) {
 Blockly.JavaScript['device_pause'] = function(block) {
     // Pause statement.
     var pause = Blockly.JavaScript.valueToCode(block, 'PAUSE', Blockly.JavaScript.ORDER_ASSIGNMENT) || '100';
-    return 'basic.pause(' + pause + ');\n';
+    return 'pauseHelper(' + pause + ');\n';
```
Comment (bfjelds) [inline IoTBlockly/IoTBlocklyBackgroundApp/public/blockly/javascript-custom.js:15]: is pauseHelper defined in uncommented code somewhere?

File: IoTBlockly/IoTBlocklyBackgroundApp/StartupTask.cs (hunk 2 above)
Comment (alecont) [inline IoTBlockly/IoTBlocklyBackgroundApp/StartupTask.cs]: ah my bad... left in this test code... i'll clean up. pauseHelper is defined in the .js/.html files themselves

File: IoTBlockly/IoTBlocklyBackgroundApp/public/blockly/javascript-custom.js (hunk 3 above)
Comment (alecont) [inline IoTBlockly/IoTBlocklyBackgroundApp/public/blockly/javascript-custom.js:15]: same as above. pauseHelper is injected into the code that comes across from index.html, so we're ok here

**Pull request merged**